File: core/src/com/jediterm/core/Platform.java
package com.jediterm.core;

import org.jetbrains.annotations.NotNull;

import java.util.Locale;

public enum Platform {
  Windows,
  OS2,
  Mac,
  Linux,
  Other;

  private static @NotNull String getOsNameLowerCase() {
    return System.getProperty("os.name").toLowerCase(Locale.ENGLISH);
  }

  public static @NotNull Platform current() {
    String osName = getOsNameLowerCase();
    if (osName.startsWith("windows")) return Platform.Windows;
    if (osName.startsWith("os/2") || osName.startsWith("os2")) return Platform.OS2;
    if (osName.startsWith("mac")) return Platform.Mac;
    if (osName.startsWith("linux")) return Platform.Linux;
    return Platform.Other;
  }
}

File: core/bin/main/com/jediterm/core/Platform.kt
package com.jediterm.core

internal enum class Platform {
  Windows,

  @Suppress("EnumEntryName")
  macOS,
  Linux,
  Other;

  companion object {
    private var CURRENT: Platform = detectCurrent()

    @JvmStatic
    fun current(): Platform = CURRENT

    @JvmStatic
    fun isWindows(): Boolean = CURRENT == Windows

    @JvmStatic
    fun isMacOS(): Boolean = CURRENT == macOS

    private fun detectCurrent(): Platform {
      val osName = System.getProperty("os.name").lowercase()
      if (osName.startsWith("windows")) return Windows
      if (osName.startsWith("mac")) return macOS
      if (osName.startsWith("linux")) return Linux
      return Other
    }
  }
}
